node_test/integration_test.go

```diff
@@ -216,7 +216,9 @@ func RunIntegrationTestCase(tc TestCase, t *testing.T) {
 
 		var chainLastConfirmedBlockNum uint64
 		if infra.mockChain != nil {
+			infra.mockChain.BlockNumMu.Lock()
```

Comment:
This is a code smell that an abstraction is broken since the test is reaching into the chain and manipulating the mutex.

Reply:
Removed the direct use of this mutex from the tests. I believe https://github.com/statechannels/go-nitro/pull/1654 solved the issue that this mutex was trying to fix.